homeassistant/components/whirlpool/sensor.py

```diff
@@ -105,13 +105,15 @@ class WhirlpoolSensorEntityDescription(
     WhirlpoolSensorEntityDescription(
         key="state",
         name="State",
-        icon=ICON_W,
+        translation_key="whirlpool",
+        device_class=SensorDeviceClass.ENUM,
         value_fn=washer_state,
     ),
     WhirlpoolSensorEntityDescription(
         key="DispenseLevel",
         name="Detergent Level",
-        icon=ICON_W,
+        translation_key="whirlpool",
```

Comment:
I recommend using one key per enum set of strings.

Reply:
I believe this is what you had in mind....